index/index.go

```diff
@@ -955,15 +965,117 @@ func (t *serializedStringTuples) At(i int) ([]string, error) {
 	return res, nil
 }
 
-// DecoderV1 provides decoding methods for the v1 index file format.
+// Decoder provides decoding methods for the v1 and v2 index file format.
 //
 // It currently does not contain decoding methods for all entry types but can be extended
 // by them if there's demand.
+type Decoder interface {
+	lookupSymbol(o uint32) (string, error)
+	SetSymbolTable(t map[uint32]string)
+	Postings(b []byte) (int, Postings, error)
+	Series(b []byte, lbls *labels.Labels, chks *[]chunks.Meta) error
+}
```

Comment:
We can probably skip making this an interface if `v2` and thus `DecoderV1` vs `DecoderV2` is not necessary.
But generally, exported interfaces should not have any unexported methods.